R/params.R

```diff
@@ -48,31 +48,54 @@ params_label <- function(inputControlFn, param) {
   label
 }
 
-params_value_to_ui <- function(inputControlFn, value) {
+params_value_to_ui <- function(inputControlFn, value, showDefault) {
+  if (is.null(showDefault)) {
+    showDefault <- TRUE
+  }
+
+  isNumericInput <- identical(inputControlFn, shiny::numericInput) ||
+    identical(inputControlFn, shiny::sliderInput)
+
   if (identical(inputControlFn, shiny::fileInput)) {
     NULL
   } else if (identical(inputControlFn, shiny::textInput)) {
     ## TODO: if long input, maybe truncate textInput values for display
 
-    classes <- class(value)
-    if ("POSIXct" %in% classes) {
-      as.character(value)
+    if (showDefault){
+      classes <- class(value)
+      if ("POSIXct" %in% classes) {
+        as.character(value)
+      } else {
+        value
+      }
     } else {
-      value
+      NULL
     }
   } else if (is.null(value)) {
-    numerics <- c(shiny::numericInput,
-                  shiny::sliderInput)
     # The numerics can't deal with a NULL value, but everything else is fine.
-    if (identical(inputControlFn, shiny::numericInput) ||
-        identical(inputControlFn, shiny::sliderInput)) {
+    if (isNumericInput) {
       0
     } else {
       value
     }
   } else {
-    ## A type/control that doesn't need special handling; just emit the value.
-    value
+    if (showDefault){
+      ## A type/control that doesn't need special handling; just emit the value.
+      value
+    } else {
+      if (isNumericInput){
+        0
```

Comment:
This might be an existing bug, but if the numeric inputs have range specifications, we should ensure that the default value (both here and above) is within the range.